random/rand.mbt

```diff
@@ -0,0 +1,232 @@
+// Copyright 2024 International Digital Economy Academy
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Currently we only support [chacha8] as the source of randomness.
+struct Rand {
+  src : ChaCha8
+}
+
+/// Create a new random number generator with [seed].
+/// @alert unsafe "Panic if seed is not 32 bytes long"
+pub fn Rand::new(~seed : Bytes = b"ABCDEFGHIJKLMNOPQRSTUVWXYZ123456") -> Rand {
+  if seed.length() != 32 {
+    abort("seed must be 32 bytes long")
+  }
+  let src = ChaCha8::new(seed)
```

Comment:
no need panic? you can always make it length 32,right?

Reply:
Yes it is possible, but another benefits for using chacha as random source is that it is cryptographically secure.

If we just pad it to 32, it no longer hold this property.